aiida/tools/importexport/common/archive.py

```diff
@@ -102,14 +198,56 @@ def filepath(self):
         """
         return self._filepath
 
+    @property
```

Comment:
See comments below. If this attribute is and only should be used internally, there is no need for a property.